ipaserver/plugins/cert.py

```diff
@@ -860,6 +861,23 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
         )
 
 
+def _emails_are_valid(cert_emails, principal_emails):
+    """
+    Checks if any email addr from DN does not appear in ldap entry,
+    comparing the domain part case-insensitively.
+    """
+
+    def lower_domain(email):
+        return email.split('@')[0] + '@' + email.split('@')[1].lower()
```

Comment:
The case where the input has multiple '@' characters should be accounted for.  Otherwise we
could end up accepting an invalid value.  We must also handle the case where input does not
have an '@' character.  Obviously this is not a valid email but the current code would raise
`IndexError`.

Also we should avoid doing the `split` multiple times.